### PR TITLE
Akotadi/create-leetcode-handler

### PR DESCRIPTION
**Pull request by akotadi**
# Description

Add a new handler for leetcode requests, currently only use two methods to retrieve the user profile and the daily challenge. Confirmed the queries using the [leetcode-playground](https://akotadi.github.io/leetcode-playground/).
The schema used was retrieved using reverse-engineering from the queries used on the leetcode website.

# Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# How Has This Been Tested?

Manually tested and added unit tests for it.

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: otter_welcome_buddy/graphql/handlers/gql_leetcode_handler.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            "username": username,
+        }
+
+        async with LeetcodeGqlConn() as gql_conn:
+            response = await gql_conn.execute(USER_PROFILE_QUERY, variables)
+
+        if response is None:
+            return None
+        matched_user: dict[str, Any] | None = response.get("matchedUser")
+        return matched_user
```
Comment (GoberInfinity) [inline otter_welcome_buddy/graphql/handlers/gql_leetcode_handler.py:32]: `matched_user` Is not being used anywhere else, maybe returning `response.get("matchedUser")` is a better idea?

File: otter_welcome_buddy/graphql/handlers/gql_leetcode_handler.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+        Returns:
+            dict[str, Any]: The daily challenge data if found, otherwise None.
+        """
+        async with LeetcodeGqlConn() as gql_conn:
+            response = await gql_conn.execute(ACTIVE_DAILY_CHALLENGE_QUERY)
+
+        if response is None:
+            return None
+        daily_challenge: dict[str, Any] | None = response.get("activeDailyCodingChallengeQuestion")
```
Comment (GoberInfinity) [inline otter_welcome_buddy/graphql/handlers/gql_leetcode_handler.py:47]: Same comment as the one in the lines above.

File: otter_welcome_buddy/graphql/schema/leetcode.gql [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+  submitStatsGlobal: UserSubmitStatsNode
+  submitStats: UserSubmitStatsNode
+  activeBadge: UserBadgeNode
+  contestBadge: UserBadgeNode
+  tagProblemCounts: TagProblemCountsCategoryNode!
+  languageProblemCount: [LanguageProblemCountNode!]!
+  isDiscussAdmin: Boolean!
+  isDiscussStaff: Boolean!
+  isActive: Boolean!
+  # nameColor - nullable
```
Comment (GoberInfinity) [inline otter_welcome_buddy/graphql/schema/leetcode.gql:97]: Any reason why this line was commented but no explanation of why we should keep it in the codebase?

File: otter_welcome_buddy/graphql/schema/leetcode.gql [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+}
+
+type CodingChallengeNodeV2 {
+  challenges: [DailyChallengeNodeV2!]!
+  weeklyChallenges: [DailyChallengeNodeV2!]!
+}
+
+type Query {
+  matchedUser(username: String!): UserNode
+  """"""
```
Comment (GoberInfinity) [inline otter_welcome_buddy/graphql/schema/leetcode.gql:327]: Maybe removing these extra `""""""` or why they're needed?

**Review by GoberInfinity — APPROVED** [review]

File: otter_welcome_buddy/graphql/handlers/gql_leetcode_handler.py (hunk 1 above)
Comment (akotadi) [reply otter_welcome_buddy/graphql/handlers/gql_leetcode_handler.py:32]: Unfortunately the linter complains, so is mainly for typing reasons

File: otter_welcome_buddy/graphql/handlers/gql_leetcode_handler.py (hunk 2 above)
Comment (akotadi) [reply otter_welcome_buddy/graphql/handlers/gql_leetcode_handler.py:47]: Same command as above

File: otter_welcome_buddy/graphql/schema/leetcode.gql (hunk 3 above)
Comment (akotadi) [reply otter_welcome_buddy/graphql/schema/leetcode.gql:97]: I keep it for reference, is a field but not sure why is referring to

File: otter_welcome_buddy/graphql/schema/leetcode.gql (hunk 4 above)
Comment (akotadi) [reply otter_welcome_buddy/graphql/schema/leetcode.gql:327]: Is just to make it more readable

**Pull request merged**